### ISSUE TITLE
feature req: ignore birthday for selected people

### ISSUE THREAD

**Issue by nutpantz** [m-i-n-a-r/birday#273]
I have birthdays for just about everyone, however a few i don't care to be reminded about on the day, but i also don't want to delete their birthday which is the only other option.

It would be nice to have a "do not show" or "ignore" option in the app. The default being to be reminded of every birthday. 

This is just for the reminder on the birthday it self, everything else needs not change.

**Comment by m-i-n-a-r** [comment]
You can manage this using the favorites and the settings for them! Just add the events you want to be reminded of to your favorites and enable the "notification only for favorites" option 😄

**Comment by nutpantz** [comment]
So you are suggesting that if i have 150 contacts and 10 i don't want to see notifications for, i should mark 140 as favorites?
That seems a little extreme, plus it makes it so that if i have favorites they get lost just to ignore a few.

**Comment by m-i-n-a-r** [comment]
Well you got a point, but I'm not sure on how to implement this feature while keeping the app easy to use. I'll think about it :)

**Comment by nutpantz** [comment]
I would suggest having 3 options for the favorites button. Favorites -normal - ignore

**Comment by herrbrixter** [comment]
I agree with @nutpantz. I prefer an ignore option because I only want to ignore a few persons' birthdays.

**Comment by m-i-n-a-r** [comment]
Ok, I'll try to work on this feature this month!

**Comment by m-i-n-a-r** [comment]
> > I would suggest having 3 options for the favorites button. Favorites -normal - ignore
> 
> Totally agree. I have 100+ contacts with birthdays, but I want to get notifications for ~30 of them on their birthday and 2 weeks before (favorites), ~30 only on their birthday (normal), and the rest should be ignored (I just want to see them on the homepage).
> 
> I've had this exact functionality in Google Calendar for about 5 years now in the form of lists: birthdays from contacts (ignored, no priority), regular birthdays (low priority), and favorite birthdays (high priority), so I like the idea of 3 options in this great app. A year ago when I first tested Birday, I uninstalled it exactly because of this. So if @m-i-n-a-r adds such functionality, I'll be over the moon (as the current GCal workflow is not very convenient due to manual editing).

I'm on it, even if I'm not sure I'll release it in the next update, but I will add it. An important question is: given that the default should and must be the "normal" status, is that ok to change to favorite or ignored by pressing the star button in the home page list, just like now (and cycling between normal -> favorite -> ignored -> normal and so on)? I can't think of a faster way 😕 

Also, this will require a DB migration and I have to make sure to don't break the import backup function with previous versions of the db. If I do, many people will get reeeeally angry.

**Comment by para2ox** [comment]
> I'm on it, even if I'm not sure I'll release it in the next update, but I will add it.
>

> Also, this will require a DB migration and I have to make sure to don't break the import backup function with previous versions of the db. If I do, many people will get reeeeally angry.
>
Many thanks! I'll wait as long as you need 🙂
> An important question is: given that the default should and must be the "normal" status, is that ok to change to favorite or ignored by pressing the star button in the home page list, just like now (and cycling between normal -> favorite -> ignored -> normal and so on)? I can't think of a faster way 😕
> 
Yes, that's fine. Or something like a pop-up list:
![CleanShot 2024-04-05 at 14 29 01@2x](https://github.com/m-i-n-a-r/birday/assets/44606291/0d1866df-98b9-412d-a0eb-473d573589fc)

**Comment by herrbrixter** [comment]
I'm good with "cycling between normal -> favorite -> ignored -> normal".

What's the difference between favorite and normal? In my case, I just have 2 groups of people: people I want to greet and people I don't want to greet.

**Comment by para2ox** [comment]
> What's the difference between favorite and normal? In my case, I just have 2 groups of people: people I want to greet and people I don't want to greet.

- Favorites — people whose birthday I want to prepare for in advance (gifts, joint plans, etc.)
- Normal — people I just want to congratulate with words.

**Comment by nutpantz** [comment]
Favorites- people who i have to mention it's their birthday on their birthday or they pout, like little girls under 10

Normal- people who i have to congratulate it's their birthday some time that week, mostly people who think they are important

Ignore- people who always forget my birthday.

**Comment by m-i-n-a-r** [comment]
Not yet, I'll probably work on it in the next few weeks. I had to release the update to fix the android 16 crash first :(

**Comment by nutpantz** [comment]
Was this ever in a release?
